[NO GBP]Supermatter zap power generation takes perspective of the machines subsystem.

Pickle-Coding opened the pull request:
## About The Pull Request
The power generation of supermatter zaps will scale by the perceived time passage of the machines subsystem, rather than the world. This will prevent the supermatter from outputting a higher power in the machines subsystem perspective when the machines subsystem lag than what the supermatter is reporting. Also removes the delta time limit.

Testmerge this first, because I suck at testing lag.
## Why It's Good For The Game
This will make the power transmission from what NT CIMS report more accurate regardless of SSmachines lag, which should reduce confusion. The vast, vast majority of power consumption scales by the machines tick rate, so it doesn't make sense to let the supermatter be more effective for power due to machines lag. Actually makes it possible to balance supermatter power generation and makes it far more meaningful without having to worry about lag.

Also subsystems taking minutes to fire because a meteor is moving through space is getting normalized, so lets just remove the delta time cap for the supermatter zaps. I think people will care more about the power generation than the 1 in a million chance someone decides to remove the grounding rods and let themselves get zapped after atmos slept for an hour.
## Changelog
:cl:
code: Supermatter zap power generation takes perspective of the machines subsystem.
/:cl:

Pickle-Coding committed:
Supermatter zap power time scaling is in the SSmachines perspective.

Makes the perspective of time passed for supermatter zap power generation be on SSmachines instead of world time. This will cause the power generation (what the grid reports) to be consistent regardless of machines lag.

Also removes the maximum delta time allowed for the zaps, because subsystems taking minutes to fire due to a random meteor moving in space is getting normalized.

lessthnthree commented:
> Also subsystems taking minutes to fire because a meteor is moving through space is getting normalized, so lets just remove the > 1 in a million chance someone decides to remove the grounding rods and let themselves get zapped after atmos slept for an hour.

Feature.

Pickle-Coding committed:
What.

I have no idea how the delta_time var got lost. Must have hit ctrl-z or something somehow before commiting.

Jacquerel commented:
Please pester someone on discord (ping @keyholders) to testmerge this if you believe that is a requirement for merging it.
